From 4a89fc9e15b7d5e1cdb2d01be2598670c479595f Mon Sep 17 00:00:00 2001
From: "Guoqing,Zhang" <guoqingx.zhang@intel.com>
Date: Thu, 24 Sep 2015 17:21:58 +0800
Subject: [PATCH] atomisp: Add ioctl ATOMISP_IOC_S_SENSOR_RUNMODE to set sensor
 run_mode

add ioctl ATOMISP_IOC_S_SENSOR_RUNMODE to set sensor run_mode.

Change-Id: I6cb5519a740db5279fac82b67314d8fe2a43cb38
Signed-off-by: Guoqing,Zhang <guoqingx.zhang@intel.com>
Original-Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-19013
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5541
---
 include/linux/atomisp.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/linux/atomisp.h b/include/linux/atomisp.h
index f75555c..c12d9aa 100644
--- a/include/linux/atomisp.h
+++ b/include/linux/atomisp.h
@@ -895,6 +895,12 @@ struct atomisp_acc_fw_load_to_pipe {
 	__u32 type;			/* Binary type */
 	__u32 reserved[3];		/* Set to zero */
 };
+/*
+ * Set Senor run mode
+ */
+struct atomisp_s_runmode {
+	__u32 mode;
+};
 
 #define ATOMISP_ACC_FW_LOAD_FL_PREVIEW		(1 << 0)
 #define ATOMISP_ACC_FW_LOAD_FL_COPY		(1 << 1)
@@ -1212,6 +1218,8 @@ struct atomisp_sensor_ae_bracketing_lut {
 #define ATOMISP_IOC_S_SENSOR_EE_CONFIG \
 	_IOW('v', BASE_VIDIOC_PRIVATE + 47, unsigned int)
 
+#define ATOMISP_IOC_S_SENSOR_RUNMODE \
+	_IOW('v', BASE_VIDIOC_PRIVATE + 48, struct atomisp_s_runmode)
 /*
  * Reserved ioctls. We have customer implementing it internally.
  * We can't use both numbers to not cause ABI conflict.
-- 
1.9.1

